pyro/nn/module.py

```diff
@@ -409,6 +421,8 @@ def named_pyro_params(self, prefix="", recurse=True):
             yield elem
 
     def _pyro_set_supermodule(self, name, context):
+        if _is_module_local_param_enabled():
+            context.param_state.update(self._pyro_context.param_state)
```

Comment:
This ensures values of vanilla `pyro.param` statements are preserved by nesting of `PyroModule` instances. This update could probably be refined to include only the vanilla values, but that's just an optimization.

Reply:
Updated further to only include the vanilla `param` values.